libvast/src/arrow_table_slice_builder.cpp

```diff
@@ -108,7 +108,8 @@ create_table_slice(const std::shared_ptr<arrow::RecordBatch>& record_batch,
   return result;
 }
 
-bool verify_record_batch(const arrow::RecordBatch& record_batch) {
+[[maybe_unused]] bool
+verify_record_batch(const arrow::RecordBatch& record_batch) {
```

Comment:
Note to the reviewer: We only used this in an assertion below, and clang-15 now warns in builds without assertions that this function is unused.